connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneConfigTest.java

```diff
@@ -0,0 +1,125 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime.standalone;
+
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+
+public class StandaloneConfigTest {
+
+    private static final String HTTPS_LISTENER_PREFIX = "listeners.https.";
+
+    @Test
+    public void testRestServerPrefixedSslConfigs() {
+        Map<String, String> httpsListenerProps = new HashMap<String, String>() {
+            {
+                put(HTTPS_LISTENER_PREFIX + SslConfigs.SSL_KEY_PASSWORD_CONFIG, "ssl_key_password");
+                put(HTTPS_LISTENER_PREFIX + SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, "ssl_keystore");
+                put(HTTPS_LISTENER_PREFIX + SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, "ssk_keystore_password");
+                put(HTTPS_LISTENER_PREFIX + SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, "ssl_truststore");
+                put(HTTPS_LISTENER_PREFIX + SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, "ssl_truststore_password");
+            }
+        };
+
+        Set<String> passwordConfigs = Stream.of(
+            SslConfigs.SSL_KEY_PASSWORD_CONFIG,
+            SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG,
+            SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG
+        ).map(key -> HTTPS_LISTENER_PREFIX + key)
+            .collect(Collectors.toSet());
+
+        Map<String, Object> expectedListenerProps = httpsListenerProps.entrySet().stream()
+            .collect(Collectors.toMap(
+                entry -> entry.getKey().substring(HTTPS_LISTENER_PREFIX.length()),
+                entry -> passwordConfigs.contains(entry.getKey())
+                        ? new Password(entry.getValue())
+                        : entry.getValue()
+            ));
+
+        Map<String, String> props = new HashMap<String, String>() {
+            {
+                // Base props required for standalone mode
+                put(WorkerConfig.KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+                put(WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+                put(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG, "/tmp/foad");
+
+                // Custom props for test
+                putAll(httpsListenerProps);
+            }
+        };
```

Comment:
Wouldn't it be simpler and easier to understand the differences between the tests by pulling out the common objects? The whole class would then be something like:
```
    private static final String HTTPS_LISTENER_PREFIX = "listeners.https.";

    private static final Map<String, Object> EXPECTED;
    static {
        Map<String, Object> settings = new HashMap<>();
        settings.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, "ssl_key_password");
        settings.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, "ssl_keystore");
        settings.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, new Password("ssl_keystore_password"));
        settings.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, "ssl_truststore");
        settings.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, new Password("ssl_truststore_password"));
        EXPECTED = Collections.unmodifiableMap(settings);
    }

    private static final Map<String, String> STANDALONE;
    static {
        Map<String, String> settings = new HashMap<>();
        settings.put(WorkerConfig.KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
        settings.put(WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
        settings.put(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG, "/tmp/foad");
        STANDALONE = Collections.unmodifiableMap(settings);
    }

    private static Map<String, String> withStringValues(Map<String, ?> inputs, String prefix) {
        return inputs.entrySet().stream()
                     .collect(Collectors.toMap(
                              entry -> prefix + entry.getKey(),
                              entry -> literalValue(entry.getValue())
                      ));
    }

    private static String literalValue(Object value) {
        return value instanceof Password ? ((Password)value).value() : value.toString();
    }

    @Test
    public void testRestServerPrefixedSslConfigs() {
        Map<String, String> props = new HashMap<>(STANDALONE);
        props.putAll(withStringValues(EXPECTED, HTTPS_LISTENER_PREFIX));

        StandaloneConfig config = new StandaloneConfig(props);
        assertEquals(EXPECTED, config.valuesWithPrefixAllOrNothing(HTTPS_LISTENER_PREFIX));
    }

    @Test
    public void testRestServerNonPrefixedSslConfigs() {
        Map<String, String> props = new HashMap<>(STANDALONE);
        props.putAll(withStringValues(EXPECTED, ""));

        StandaloneConfig config = new StandaloneConfig(props);
        Map<String, Object> actualProps = config.valuesWithPrefixAllOrNothing(HTTPS_LISTENER_PREFIX)
                                                .entrySet().stream()
                                                .filter(entry -> EXPECTED.containsKey(entry.getKey()))
                                                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
        assertEquals(EXPECTED, actualProps);
    }
```

Reply:
I like a lot of this and will incorporate it. I'd rather not declare constants like `STANDALONE` and `EXPECTED` just because they seem pretty specific to these two tests and may not be applicable to other tests that get added to this class, but pretty much everything else (especially the handling of `Password`-type configs) is great.

Reply:
Actually, it looks like the `ConfigDef` class already has some logic to convert parsed types (including `Password`s) into strings: https://github.com/apache/kafka/blob/f35e6496aa003dc0667dc338001a5b51be78fe76/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L761-L783

So we can even just use that and remove the `literalValue` method